yaptide/routes/celery_routes.py

```diff
@@ -156,19 +159,19 @@ def delete(user: UserModel):
                                     })
 
         tasks = fetch_celery_tasks_by_sim_id(sim_id=simulation.id)
-
-        celery_ids = [task.celery_id for task in tasks]
-
-        result: dict = cancel_job(merge_id=simulation.merge_id, celery_ids=celery_ids)
-
-        if "merge" in result:
-            update_simulation_state(simulation=simulation, update_dict=result["merge"])
-            for i, task in enumerate(tasks):
-                update_task_state(task=task, update_dict=result["tasks"][i])
-
-            return yaptide_response(message="", code=200, content=result)
-
-        return error_internal_response()
+        celery_ids = [
+            task.celery_id for task in tasks
+            if task.task_state in [EntityState.PENDING.value, EntityState.RUNNING.value, EntityState.UNKNOWN.value]
+        ]
+        celery_app.control.revoke(simulation.merge_id, terminate=True, signal="SIGINT")
+        celery_app.control.revoke(celery_ids, terminate=True, signal="SIGINT")
+        update_simulation_state(simulation=simulation, update_dict={"job_state": EntityState.CANCELED.value})
+        for task in tasks:
+            if task.task_state in [EntityState.PENDING.value, EntityState.RUNNING.value]:
+                update_task_state(task=task, update_dict={"task_state": EntityState.CANCELED.value})
+
+        terminate_unfinished_tasks.delay(celery_ids=celery_ids, simulation_id=simulation.id)
```

Comment:
is that blocking or non-blocking ?
If non-blocking then why do you later return with "Cancelled successfully" message ?

Reply:
Maybe another state would need to be introduced (in separate issue), like "CANCELLING" which would serve the case of long-running-cancellation state.

Reply:
It's non-blocking. I assume that function executes successfully. Testing has proofed that even with large number of tasks and large number of particles it works. I added some timeout there to wait until tasks get canceled after first revoke before trying again.

Reply:
I personally believe that it doesn't matter for frontend user that cancelling will last some time in background after clicking button. But if there is or will be some backend part dependent on it then yes.